realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -602,6 +603,28 @@ public void onResult(int count) {
                         }
                     }
                 }
+                File managementFolder = new File(realmFolder, realmFileName + management);
+                if (managementFolder.exists()) {
+                    filesToDelete = Arrays.asList(
+                            new File(managementFolder, realmFileName + ".lock"),
+                            new File(managementFolder, realmFileName + ".log_a"),
+                            new File(managementFolder, realmFileName + ".log_b"),
+                            new File(managementFolder, realmFileName + ".log"));
+                    for (File fileToDelete : filesToDelete) {
+                        if (fileToDelete.exists()) {
+                            boolean deleteResult = fileToDelete.delete();
+                            if (!deleteResult) {
+                                realmDeleted.set(false);
+                                RealmLog.w("Could not delete the file " + fileToDelete);
+                            }
+                        }
+                    }
+                    boolean deleteResult = managementFolder.delete();
```

Comment:
Oh yes, that's a valid point. But the management folder's name is derived from the Realm file's name and you will only have a collision if two Realms have the same name - and then the two Realms are the same and we only have one :-)